coffee/shepherd.coffee

```diff
@@ -117,6 +117,9 @@ class Step extends Evented
     @tether = new Tether extend(tetherOpts, @options.tetherOptions)
 
   show: =>
+      @options.beforeShowPromise?()?.then(=> @_show()) ? @_show()
```

Comment:
Nice. Mind simplifying `.then(=> @_show())` to `.then(@_show)`? In this case they're functionally the same.